### PR TITLE
IA-1742: Page edition users dropdown error

### PR DESCRIPTION
**Pull request by beygorghor**
Previous code was removing current user from users dropdown options while displaying list of allowed users that can access a page. 
Screenshot 2023-01-09 at 12 49 36

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes

Refactor users selection using renderTags and disable user deletion on current user.

## How to test

Edit a page.

Check the Ui in not displaying an error, try to delete yourself in the users dropdown

## Print screen / video

https://user-images.githubusercontent.com/12494624/211302038-3afb23a9-8d24-4d12-af3b-7581f47165c2.mov

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    ...props
+} = {}) => {
+    const value = field?.value || [];
+    const currentUser = useCurrentUser();
+    const { data, isFetching: isFetchingProfiles } = useGetProfiles();
+    const profiles = data ? data.profiles : [];
+    const profilesList = profiles.map(p => ({
+        value: p.user_id,
+        label: getDisplayName(p),
+    }));
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx]: I think profilesList should be memoized:
```javascript
const profilesList = useMemo(()=>{
     if(!data) return [];
     return data.profiles.map(p=>({// etc
     });
}, [data]);
```

File: hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const handleChange = useCallback(
+        (newValue: string): void => {
+            const fieldValue: string[] | undefined = newValue
+                ? newValue.split(',')
+                : undefined;
+            if (field && form) {
+                // Disable delete of current user chip
+                if (fieldValue?.includes(`${currentUser.user_id}`)) {
+                    form.setFieldValue(field.name, fieldValue);
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx]: This is not super robust: The selection will only show the current user has selected themselves first. Maybe we should preselect the current user:

https://user-images.githubusercontent.com/38907762/211318257-3cf2c9ac-ca79-4c4a-b67f-bc229e91c666.mov

**Review by quang-le — CHANGES_REQUESTED** [review]
just a small glitch remaining , see comments

File: hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx]: yes i forgot the creation my bad !

**Review by codeKameleon — APPROVED** [review]

**Pull request merged**